ggshield/utils/archive.py

```diff
@@ -0,0 +1,124 @@
+import shutil
+import stat
+import tarfile
+from pathlib import Path
+from typing import List, Optional, Tuple
+from zipfile import ZipFile, ZipInfo
+
+
+class UnsafeArchive(Exception):
+    """
+    Raised when an archive is considered unsafe
```

Comment:
Might be worth being a bit more specific ? 
```suggestion
    Raised when an archive is considered unsafe: if it could cause to write/overwrite or link outside the target folder when unpacking.
```